### PR TITLE
Add webhook sample infrastructure

### PR DESCRIPTION
**Pull request by pakrym-stripe**
Adds some conventions around webhook examples and  a way to test them.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,6 +2,7 @@
 update-version:
 	@echo "$(VERSION)" > VERSION
 	@perl -pi -e 's|"version": "[.\-\d\w]+"|"version": "$(VERSION)"|' package.json
+	@find . -name 'package.json' -exec perl -pi -e 's|"stripe": "[\^\d\w.]*?"|"stripe": "^$(VERSION)"|' {} +
```
Comment (pakrym-stripe) [inline Makefile:5]: Keep sample version in sync

File: test/Integration.spec.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+import * as childProcess from 'child_process';
+
+const testUtils = require('../testUtils');
+
+describe('Integration test', function() {
+  this.timeout(30000);
+  const testExec = (cmd: string): Promise<void> => {
```
Comment (pakrym-stripe) [inline test/Integration.spec.ts:9]: run processes async so the test runner can actually cancel them if hanging.

File: examples/webhook-signing/express/main.ts [hunk 3]
```diff
@@ -65,6 +68,9 @@ app.post(
   }
 );
 
-app.listen(3000, (): void => {
-  console.log('Example app listening on port 3000!');
-});
+const server = app.listen();
+console.log(
+  `Webhook endpoint available at http://localhost:${
+    (<AddressInfo>server.address()).port
```
Comment (pakrym-stripe) [inline examples/webhook-signing/express/main.ts:74]: Use random ports so things don't collide on CI

File: test/Integration.spec.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    const child = childProcess.exec(cmd);
+
+    child.stdout?.pipe(process.stdout);
```
Comment (anniel-stripe) [inline test/Integration.spec.ts]: Thanks for improving these tests!

File: examples/webhook-signing/koa/package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "webhook-signing-example-koa",
+  "version": "1.0.0",
+  "description": "",
+  "main": "./main.ts",
+  "scripts": {
+    "run": "ts-node-transpile-only ./main.ts",
+    "prepare": "../prepare.sh"
```
Comment (anniel-stripe) [inline examples/webhook-signing/koa/package.json:8]: Where is this used?

File: examples/webhook-signing/koa/main.ts [hunk 6]
```diff
@@ -56,10 +57,10 @@ app.use(async (ctx, next) => {
   if (ctx.request.path === '/webhook') {
     return handleWebhook(ctx, next);
   }
-  const name = ctx.request.body?.name ?? 'world';
+  const name = ctx.request.body?.['name'] ?? 'world';
```
Comment (anniel-stripe) [inline examples/webhook-signing/koa/main.ts:61]: This syntax is new to me, is this purposeful?

File: examples/webhook-signing/koa/package.json (hunk 5 above)
Comment (pakrym-stripe) [reply examples/webhook-signing/koa/package.json:8]: Runs when `npm install` is executed

File: examples/webhook-signing/koa/main.ts (hunk 6 above)
Comment (pakrym-stripe) [reply examples/webhook-signing/koa/main.ts:61]: yeah, ctx.request.body is of `unknown` type. You can't dot into `unknown` but you can index.

**Review by anniel-stripe — APPROVED** [review]

**Pull request merged**